homeassistant/components/synology_dsm/config_flow.py

```diff
@@ -368,6 +397,43 @@ async def async_step_2sa(
 
         return await self.async_step_user(user_input)
 
+    async def async_step_backup_share(
+        self, user_input: dict[str, Any], errors: dict[str, str] | None = None
+    ) -> ConfigFlowResult:
+        """Select backup location."""
+        if TYPE_CHECKING:
+            assert self.shares is not None
+
+        if not self.saved_user_input:
+            self.saved_user_input = user_input
+
+        if CONF_BACKUP_PATH not in user_input and CONF_BACKUP_SHARE not in user_input:
```

Comment:
It's enough to check if user input is falsy. Both items are required.

Reply:
- #136655